src/libsyntax/parse/lexer/mod.rs

```diff
@@ -269,6 +281,15 @@ impl<'a> StringReader<'a> {
         Self::push_escaped_char_for_msg(&mut m, c);
         self.fatal_span_(from_pos, to_pos, &m[..])
     }
+
+    fn struct_span_fatal(&self,
```

Comment:
Is this method necessary after the last change?

Reply:
Still yes.

Reply:
Isn't it being used only in `fail_unterminated_raw_string` above? The span for the error and the span_label could be made only once (`self.mk_span(pos, pos)`) and the message and label literal could be the same variable.